test/arrayops.jl

```diff
@@ -362,6 +362,14 @@ a = [0,1,2,3,0,1,2,3]
 @test findprev(isodd, [2,4,5,3,9,2,0], 7) == 5
 @test findprev(isodd, [2,4,5,3,9,2,0], 2) == 0
 
+# find with general iterables
+s = "julia"
+@test find(s) == [1,2,3,4,5]
```

Comment:
Potential solutions:

- change `a != 0` to `!isequal(a, 0)`, won't find -0.0
- change `a != 0` to `!(isequal(a, 0) | isequal(a, -0.0))`, weird and maybe inefficient
- change `a != 0` to `a != convert(eltype(A), 0)`